### DISCUSSION TITLE
How to make requests to GoTrue endpoints?

### DISCUSSION THREAD

**Discussion by mayank99** [supabase/supabase#7750] [Questions] [answered]
I'm trying to implement my own server-side version of [`getUserByCookie`](https://github.com/supabase/gotrue-js/blob/09c0ce0721ee61f515889ddb7cb5d2de4e57b5ee/src/GoTrueApi.ts#L697) because the official version only works for Next/Express. I've passed the access token from client to my server but I'm unable to fetch the user info with that token.

I'm making a `GET` request to `<project-id>.supabase.co/auth/v1/user` with `Authorization: Bearer {token}` in my headers, but I keep getting a 401 response with the message "No API key found in request". 

I've also tried `/auth/v1/settings` with my anon key but I get the same response.

What am I doing wrong? Is it the url? Do I need to specify an additional header?

**Comment by GaryAustin1** [comment, marked as answer]
Supabase-js also sets this header for all calls. 
headers['apikey'] = this.supabaseKey

**Reply by mayank99** [reply]
Ah ok so it needs both `apikey` (supabase anon key) and `Authorization` (user's jwt). Thanks.

Is this documented somewhere?

**Reply by GaryAustin1** [reply]
I'm sure in a couple of spots, but not in github gotrue readme (at least at a quick glance)...
The API tab in the Dashboard shows the headers for Auth...

![image](https://user-images.githubusercontent.com/54564956/179541256-46e7fd03-8632-4626-be8f-fa31ee779047.png)
